From df2a262dde05364356022c1ecad72404dcfcf512 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hannes=20M=C3=BCller?= <>
Date: Mon, 1 Dec 2025 19:48:05 +0100
Subject: [PATCH] Avoid warnings when Guile headers are included in a project
 using the compile option -Wsign-conversion.

* libguile/array-handle.h (scm_array_handle_ref, scm_array_handle_set):
Explicitly cast signed index 'p' to size_t. This is safe because
bounds checking is performed immediately prior.
* libguile/arrays.h (scm_i_raw_array): Explicitly cast 'ndim' to
uint32_t to match the signature of scm_words.
---
 libguile/array-handle.h | 4 ++--
 libguile/arrays.h       | 3 ++-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/libguile/array-handle.h b/libguile/array-handle.h
index 6ad80eb41..b43b172a5 100644
--- a/libguile/array-handle.h
+++ b/libguile/array-handle.h
@@ -105,7 +105,7 @@ scm_array_handle_ref (scm_t_array_handle *h, ssize_t p)
     /* catch overflow */
     scm_out_of_range (NULL, scm_from_ssize_t (p));
   /* perhaps should catch overflow here too */
-  return h->vref (h->vector, h->base + p);
+  return h->vref (h->vector, h->base + (size_t) p);
 }
 
 SCM_INLINE_IMPLEMENTATION void
@@ -115,7 +115,7 @@ scm_array_handle_set (scm_t_array_handle *h, ssize_t p, SCM v)
     /* catch overflow */
     scm_out_of_range (NULL, scm_from_ssize_t (p));
   /* perhaps should catch overflow here too */
-  h->vset (h->vector, h->base + p, v);
+  h->vset (h->vector, h->base + (size_t) p, v);
 }
 
 #endif
diff --git a/libguile/arrays.h b/libguile/arrays.h
index 5457ddb95..378fb019b 100644
--- a/libguile/arrays.h
+++ b/libguile/arrays.h
@@ -106,7 +106,8 @@ typedef struct scm_t_array_dim
 static inline SCM
 scm_i_raw_array (int ndim)
 {
-  return scm_words (((scm_t_bits) ndim << 17) + scm_tc7_array, 3 + ndim*3);
+  return scm_words (((scm_t_bits) ndim << 17) + scm_tc7_array,
+                    3 + (uint32_t)ndim*3);
 }
 
 SCM_INTERNAL SCM scm_i_make_array (int ndim);
-- 
2.47.3

